cmd/realm/triggers.go

```diff
@@ -69,111 +71,351 @@ var triggersDeleteCmd = &cobra.Command{
 
-	realmTriggersRes, err := realmTriggersCall.Run(astarteAPIClient)
+	deleteTriggerRes, err := deleteTriggerCall.Run(astarteAPIClient)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
-	rawRealmTriggers, _ := realmTriggersRes.Parse()
-	realmTriggers, _ := rawRealmTriggers.([]string)
-	fmt.Println(realmTriggers)
+
+	_, _ = deleteTriggerRes.Parse()
+
+	fmt.Println("ok")
 	return nil
 }
 
-func triggersShowF(command *cobra.Command, args []string) error {
-	triggerName := args[0]
+func triggersSaveF(command *cobra.Command, args []string) error {
+	if viper.GetBool("realmmanagement-to-curl") {
+		fmt.Println(`'triggers save' does not support the --to-curl option. Use 'triggers list' to get the triggers in your realm, 'triggers versions' to get their versions, and 'triggers show' to get the content of a trigger.`)
+		os.Exit(1)
+	}
 
-	getTriggerCall, err := astarteAPIClient.GetTrigger(realm, triggerName)
+	var targetPath string
+	var err error
+	if len(args) == 0 {
+		targetPath, _ = filepath.Abs(".")
+	} else {
+		targetPath, err = filepath.Abs(args[0])
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+	}
+
+	// retrieve triggers list
+	realmTriggers, err := listTriggers(realm)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 
-	utils.MaybeCurlAndExit(getTriggerCall, astarteAPIClient)
+	for _, name := range realmTriggers {
 
-	getTriggerRes, err := getTriggerCall.Run(astarteAPIClient)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+		triggerDefinition, err := getTriggerDefinition(realm, name)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		respJSON, err := json.MarshalIndent(triggerDefinition, "", "  ")
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		filename := fmt.Sprintf("/%s/%s.json", targetPath, name)
+		outFile, err := os.Create(filename)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		defer outFile.Close()
+
+		if _, err := outFile.Write(respJSON); err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+	}
+	return nil
+}
+
+func triggersSyncF(command *cobra.Command, args []string) error {
+	if viper.GetBool("to-curl") {
+		fmt.Println(`'triggers sync' does not support the --to-curl option. Install your triggers one by one with 'triggers install'.`)
 		os.Exit(1)
 	}
 
-	triggerDefinition, _ := getTriggerRes.Parse()
-	respJSON, _ := json.MarshalIndent(triggerDefinition, "", "  ")
-	fmt.Println(string(respJSON))
+	triggersToInstall := []map[string]interface{}{}
+	triggersToUpdate := []map[string]interface{}{}
+	invalidTriggers := []string{}
+
+	for _, f := range args {
+		triggerFile, err := os.ReadFile(f)
+		if err != nil {
+			return err
+		}
+		if !validateTrigger(f) {
+			invalidTriggers = append(invalidTriggers, f)
+			continue
+		}
+
+		// var astarteTrigger map[string]triggers.Trigger{}
+		var astarteTrigger map[string]interface{}
```

Comment:
I think this should be a `triggers.AstarteTrigger{}` as your above comment suggests